client/src/client_sync/v17/mining.rs

```diff
@@ -16,7 +16,7 @@ macro_rules! impl_client_v17__getblocktemplate {
         impl Client {
             pub fn get_block_template(
                 &self,
-                request: &$crate::client_sync::TemplateRequest,
+                request: &$crate::client_sync::v17::TemplateRequest,
```

Comment:
This change forces us to add a new `get_block_template` to the v29 client. Instead of doing this we can just import the type from the correct module in `v17/mod.rs`. You can see an example of how we do this for v28 and `AddressType` (`client/src/client_sync/v28/mod.rs`).